### ISSUE TITLE
High resolution image

### ISSUE THREAD

**Issue by holak0** [AlexeyAB/darknet#1057]
Hi Alex,

I want to train my Yolo with hight resolution image 2048*2048 to detect small object (Satellite image to detect car) and I need to know if I have to do any special configuration.

Thank you

**Comment by AlexeyAB** [comment]
@holak0 Hi,

If you will get error Out of memory, then to reduce memory consumption you will need to set `batch=64 subdivisions=64`

You can train with `batch=64 subdivision=64`
* `width=2048 height=2048` in `yolov3.cfg` if you have GPU with 32 GB-RAM 
* `width=1024 height=1024` in `yolov3.cfg` if you have GPU with 8 GB-RAM 
* `width=2048 height=2048` in `yolov3-tiny.cfg` if you have GPU with 4 GB-RAM 

If your satellite images has different scales or resolutions, then you should use `random=1`, else use `random=0`.

Try to do: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

especially:

> recalculate anchors for your dataset for width and height from cfg-file: `darknet.exe detector calc_anchors data/obj.data -num_of_clusters 9 -width 416 -height 416` then set the same 9 anchors in each of 3 [yolo]-layers in your cfg-file

and

> for training for small objects - set layers = -1, 11 instead of https://github.com/AlexeyAB/darknet/blob/6390a5a2ab61a0bdf6f1a9a6b4a739c16b36e0d7/cfg/yolov3.cfg#L720 and set stride=4 instead of https://github.com/AlexeyAB/darknet/blob/6390a5a2ab61a0bdf6f1a9a6b4a739c16b36e0d7/cfg/yolov3.cfg#L717

**Comment by holak0** [comment]
Hi AlexeyAB,

First thank you for your help.

To recalculate the anchor in my case, I have to write : 
**darknet.exe detector calc_anchors data/obj.data -num_of_clusters 9 -width 2048 -height 2048**

???

**Comment by AlexeyAB** [comment]
@holak0 Hi,

If you use width=1024 height=1024 in cfg-file, then you should use
`darknet.exe detector calc_anchors data/obj.data -num_of_clusters 9 -width 1024 -height 1024`

**Comment by holak0** [comment]
Hi AlexeyAB,

I have tried with Yolo V3 I get a lot of errors and because I don't have a lot time to give my project I have returned to Yolo V2 it worked for me but the Avg loss stagnates in 0.5 and 0.7 (for 5000 iterations) and I don't know why !!!!

this is my cfg file 
`[net]
# Testing
batch=64
subdivisions=8
# Training
# batch=64
# subdivisions=8
height=416
width=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 80200
policy=steps
steps=40000,60000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=64
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky


#######

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[route]
layers=-9

[convolutional]
batch_normalize=1
size=1
stride=1
pad=1
filters=64
activation=leaky

[reorg]
stride=2

[route]
layers=-1,-4

[convolutional]
batch_normalize=1
size=3
stride=1
pad=1
filters=1024
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=30
activation=linear


[region]
anchors =  1.3221, 1.73145, 3.19275, 4.00944, 5.05587, 8.09892, 9.47112, 4.84053, 11.2364, 10.0071
bias_match=1
classes=1
coords=4
num=5
softmax=1
jitter=.3
rescore=1

object_scale=5
noobject_scale=1
class_scale=1
coord_scale=1

absolute=1
thresh = .6
random=1
`

**Comment by AlexeyAB** [comment]
>  Avg loss stagnates in 0.5 and 0.7 (for 5000 iterations)

This is normal avg loss.

> I have tried with Yolo V3 I get a lot of errors 

what kind of errors did you get?

**Comment by holak0** [comment]
> This is normal avg loss
how to improve this avg loss I have to add other images for training ?? or do more iterations? I want to inform you that I use 4 images in training and in every image I have about 300 boxes.

**Comment by AlexeyAB** [comment]
Does your dataset contain only 4 images? It is recommended to use 2000 images per class.

Set `max=300` in the `[region]`-layer in your cfg-file: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

**Comment by holak0** [comment]
I use satellite imagery, every image contain a lot cars that I want to detect (between 100 and 300 cars)... so it will be a problem if I use just 4 images for training ????

**Comment by AlexeyAB** [comment]
Using 4 images - you can get low avg-loss.
But you can't get high mAP (mean average precision) for test dataset if it differs from training dataset.

**Comment by kmsravindra** [comment]
Hi @AlexeyAB,

One quick question on the response you mentioned :
> 
> If you will get error Out of memory, then to reduce memory consumption you will need to set `batch=64 subdivisions=64`
> 
> You can train with `batch=64 subdivision=64`
> 
> * `width=2048 height=2048` in `yolov3.cfg` if you have GPU with 32 GB-RAM
> * `width=1024 height=1024` in `yolov3.cfg` if you have GPU with 8 GB-RAM
> * `width=2048 height=2048` in `yolov3-tiny.cfg` if you have GPU with 4 GB-RAM
> 

Can we use multi-GPU training to reduce the RAM requirement for training high resolution images? I mean can I use eight of 4GB RAMs to train 2048 by 2048 in yolov3? OR is it a must that each GPU should be of 32 GB RAM to train such a 2048 x 2048 image in yolov3?

**Comment by AlexeyAB** [comment]
@kmsravindra Hi,

> Can we use multi-GPU training to reduce the RAM requirement for training high resolution images? I mean can I use eight of 4GB RAMs to train 2048 by 2048 in yolov3? 

No.
Each GPU should be of 32 GB RAM.
